mantis-control-plane/mantis-control-plane-server/src/test/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerTests.java

```diff
@@ -308,28 +309,28 @@ public void testGetBestFit_WithGenerationFromScaleGroup() {
 
         bestFitO =
             stateManager.findBestFit(
-                new TaskExecutorAssignmentRequest(
-                    TaskExecutorAllocationRequest.of(WORKER_ID, MACHINE_DEFINITION_1, null, 0),
+                new TaskExecutorBatchAssignmentRequest(
+                    Collections.singleton(TaskExecutorAllocationRequest.of(WORKER_ID, MACHINE_DEFINITION_1, null, 0)),
                     CLUSTER_ID));
 
         assertTrue(bestFitO.isPresent());
-        assertEquals(te5Id, bestFitO.get().getLeft());
-        assertEquals(teState5, bestFitO.get().getRight());
+        assertEquals(te5Id, bestFitO.get().getBestFit().values().stream().findFirst().get().getLeft());
+        assertEquals(teState5, bestFitO.get().getBestFit().values().stream().findFirst().get().getRight());
 
         // disable all group2 TEs and allow bestFit from group1
         teState5.onTaskExecutorStatusChange(
             new TaskExecutorStatusChange(te5Id, CLUSTER_ID, TaskExecutorReport.occupied(WORKER_ID)));
         bestFitO =
             stateManager.findBestFit(
-                new TaskExecutorAssignmentRequest(
-                    TaskExecutorAllocationRequest.of(WORKER_ID, MACHINE_DEFINITION_1, null, 0),
+                new TaskExecutorBatchAssignmentRequest(
+                    Collections.singleton(TaskExecutorAllocationRequest.of(WORKER_ID, MACHINE_DEFINITION_1, null, 0)),
                     CLUSTER_ID));
 
         assertTrue(bestFitO.isPresent());
-        assertNotEquals(te5Id, bestFitO.get().getLeft());
-        assertNotEquals(TASK_EXECUTOR_ID_1, bestFitO.get().getLeft());
+        assertNotEquals(te5Id, bestFitO.get().getBestFit().values().stream().findFirst().get().getLeft());
+        assertNotEquals(TASK_EXECUTOR_ID_1, bestFitO.get().getBestFit().values().stream().findFirst().get().getLeft());
         assertEquals(SCALE_GROUP_1,
-            Objects.requireNonNull(bestFitO.get().getRight().getRegistration())
+            Objects.requireNonNull(bestFitO.get().getBestFit().values().stream().findFirst().get().getRight().getRegistration())
```

Comment:
it would be nice to have some more UTs around the pending worker set scenarios here.